src/tasks/match_modalities/workflows/run_benchmark/main.nf

```diff
@@ -0,0 +1,131 @@
+workflow auto {
+  findStates(params, meta.config)
+  | meta.workflow.run(
+    auto: [publish: "state"]
+  )
+}
+
+workflow run_wf {
+  take:
+  input_ch
+
+  main:
+
+  // collect method list
+  methods = [
+    random_features,
+    true_features,
+    scot,
+    harmonic_alignment,
+    fastmnn,
+    procrustes
+  ]
+
+  // collect metric list
+  metrics = [
+    knn_auc,
+    mse
+  ]
+
+  output_ch = input_ch
+
+    // store original id for later use
+    | map{ id, state ->
+      [id, state + [_meta: [join_id: id]]]
+    }
+
+    // extract the dataset metadata
+    | check_dataset_schema.run(
+      fromState: [ "input": "input_mod1" ],
+      toState: { id, output, state ->
+        // load output yaml file
+        def metadata = (new org.yaml.snakeyaml.Yaml().load(output.meta)).uns
+        // add metadata from file to state
+        state + metadata
+      }
+    )
```

Comment:
Do the other inputs need to be checked ?

Reply:
this step doesn't actually check whether the input_mod1 matches the expected schema -- since the h5ad files should only have been generated by the process_dataset workflow this should always be the case.

Instead, I'm using the `check_dataset_schema`'s ability of outputting the uns as a YAML file to read in that YAML file and get to know the dataset_id and normalization_id of the given dataset.